smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/AuthSchemeResolverGenerator.kt

```diff
@@ -0,0 +1,256 @@
+package software.amazon.smithy.swift.codegen
+
+import software.amazon.smithy.aws.traits.ServiceTrait
+import software.amazon.smithy.aws.traits.auth.SigV4Trait
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait
+import software.amazon.smithy.model.knowledge.ServiceIndex
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.AuthTrait
+import software.amazon.smithy.model.traits.OptionalAuthTrait
+import software.amazon.smithy.model.traits.Trait
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.ServiceTypes
+import software.amazon.smithy.swift.codegen.utils.clientName
+import software.amazon.smithy.swift.codegen.utils.toLowerCamelCase
+
+class AuthSchemeResolverGenerator() {
+    fun render(ctx: ProtocolGenerator.GenerationContext) {
+        val rootNamespace = ctx.settings.moduleName
+        val serviceIndex = ServiceIndex(ctx.model)
+
+        ctx.delegator.useFileWriter("./$rootNamespace/${ClientRuntimeTypes.Core.AuthSchemeResolver.name}.swift") {
+            renderResolverParams(serviceIndex, ctx, it)
+            it.write("")
+            renderResolverProtocol(ctx, it)
+            it.write("")
+            renderDefaultResolver(serviceIndex, ctx, it)
+            it.write("")
+            it.addImport(SwiftDependency.CLIENT_RUNTIME.target)
+            it.addImport("AWSClientRuntime")
```

Comment:
`smithy-swift` cannot depend on `aws-sdk-swift` or its library `AWSClientRuntime`.

Even without explicitly requiring `aws-sdk-swift` in this project's `Package.swift`, adding an import of that module in codegen is creating a "backdoor" dependency (i.e. the compiler doesn't require that `aws-sdk-swift` be required as a dependency but it is still a dependency since generated code will not function without being linked against that module), and even worse this is a circular dependency, where `aws-sdk-swift` and `smithy-swift` depend on each other.

I'm not immediately sure what is the best way to break this requirement without greater research and/or discussion, but we cannot refer to `aws-sdk-swift` or any of its components from this project.

Reply:
Only dependency here was for saving values to middleware context, and luckily in following milestone (customizations milestone) I refactored it to handle it in auth scheme structs rather than in auth scheme resolver. This way auth scheme resolver (which is generic, for both smithy and AWS SDK) won't be dependent on AWSClientRuntime.

Reply:
Updated code here by deleting AWS-dependent bits, which will be moved to auth scheme in AWSClientRuntime. About re-locating entirety of SigV4 related things from sdk to smithy, I'd rather keep that out of scope for the project and have that be part of modularization project, since it will require a new round of analysis.